extensions/gitpod/src/extension.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import Log from './common/logger';
+import GitpodAuthenticationProvider from './authentication';
+import LocalApp from './localApp';
+import { enableSettingsSync, updateSyncContext } from './settingsSync';
+import { GitpodServer } from './gitpodServer';
+import TelemetryReporter from './telemetryReporter';
+
+const EXTENSION_ID = 'gitpod.gitpod-desktop';
+const ANALITYCS_KEY = 'bUY8IRdJ42KjLOBS9LoIHMYFBD8rSzjU';
+
+let telemetry: TelemetryReporter;
+
+export async function activate(context: vscode.ExtensionContext) {
+	const logger = new Log('Gitpod');
+
+	const version = vscode.extensions.getExtension(EXTENSION_ID)!.packageJSON.version;
+	telemetry = new TelemetryReporter(EXTENSION_ID, version, ANALITYCS_KEY);
+
+	/* Gitpod settings sync */
+	await updateSyncContext();
+	context.subscriptions.push(vscode.workspace.onDidChangeConfiguration(e => {
+		if(e.affectsConfiguration('gitpod')){
+			updateSyncContext();
+		}
+	}));
+
+	context.subscriptions.push(vscode.commands.registerCommand('gitpod.syncProvider.remove', async () => {
+		try {
+			await enableSettingsSync(false, telemetry);
+		} catch (e) {
+			vscode.window.showErrorMessage(`Error setting up Gitpod Code Sync Provider: ${e}`);
```

Comment:
Please log errors to output that they all captured to logs. Please review other try-catch in the desktop extension.

Reply:
Checked them all out and it seems double-logging (user facing and to output) are everywhere. Made some minor tweaks in e746a4841d10d2a255727fcf4333def8fe2faa47.